### PR TITLE
Feat/deque

### PR DESCRIPTION
**Pull request by san-ghun**
Create .c files to add functions for struct `t_sent` and struct `t_deque`.
Referencing #38 & #43 

Almost not been tested during development, NEED more tests.
Help wanted for testing.

---

\+ Add a new feature that having a working **history**
Referencing #40

### CODE REVIEW

**Comment by san-ghun** [comment]
Update the PR by adding feature of having a working **history**.

File: src/t_sent/sent_util.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	tmp = *sent;
+	while (tmp->next != NULL)
+	{
+		tmp = tmp->next;
+		len++;
+	}
+	return (len);
+}
+
+void	sent_print(t_sent *sent[])
```
Comment (miooo0o) [inline src/t_sent/sent_util.c:35]: test function! thank you!

File: src/deque/deque_delete.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+
+void	deque_del(t_deque *deque);
+t_sent	*deque_pop_front(t_deque *deque);
+t_sent	*deque_pop_back(t_deque *deque);
+
+void	deque_del(t_deque *deque)
```
Comment (miooo0o) [inline src/deque/deque_delete.c:19]: I don't think we need to malloc "deque", but it's possible to do so!

File: src/t_sent/sent_util.c
Comment (miooo0o) [inline src/t_sent/sent_util.c:1]: Thanks for the update, I'll accept the request for now for more testing.

**Review by miooo0o — APPROVED** [review]
Thanks for the update, I'll accept the request for now for more testing.

**Pull request merged**